sparse_strips/vello_api/src/execute.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2025 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Different execution modes for kernels.
+
+#[derive(Copy, Clone, Debug)]
+/// The execution mode used for the rendering process.
+pub enum ExecutionMode {
+    /// Only use scalar execution. This is recommended if you want to have
+    /// consistent results across different platforms and want to avoid unsafe code,
+    /// and is the only option if you disabled the `simd` feature. Performance will be
+    /// worse, though.
+    Scalar,
+    /// Select the best execution mode according to what is available on the host system.
+    /// This is the recommended option for highest performance.
+    #[cfg(feature = "simd")]
+    Auto,
+    /// Force the usage of neon SIMD instructions. This will lead to panics in case
+    /// the CPU doesn't support the target feature `neon`.
+    #[cfg(all(target_arch = "aarch64", feature = "simd"))]
+    Neon,
+    /// Force the usage of AVX2 SIMD instructions. This will lead to panics in case
+    /// the CPU doesn't support the target features `avx2` and `fma`.
+    #[cfg(all(target_arch = "x86_64", feature = "simd"))]
+    Avx2,
+}
```

Comment:
One alternative would be to lift this to the type-level to allow for free branching due to monomorphization, but perhaps that should be an implementation detail and not be exposed on the API. On the other hand, if we ever get something like [struct target features](https://github.com/rust-lang/rfcs/pull/3525), the consumer could guide autovectorization through monomorphization.

That said, I'd be happy to land this as-is and revisit when we have a clearer picture.